segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -385,6 +391,10 @@ private CompletableFuture<Void> claimOwnership(MetadataTransaction txn, SegmentM
         }, executor);
     }
 
+    boolean isAlwaysAtomic(SegmentMetadata segmentMetadata) {
+        return NameUtils.isAttributeSegment(segmentMetadata.getName());
```

Comment:
Are we sure that `lazyCommit` was disabled for attribute segments and system segments?

Reply:
This code is checking whether given segment is supposed to be always atomic ? Answer is yes if it is attribute segment